### PR TITLE
Credential handling

### PR DESCRIPTION
**Pull request by bbernays**
Spaces Credentials are only used when grabbing Spaces resource. So needed a way of notifying the user of only when interacting with Spaces.

Did not want to parse Errors from S3/Spaces because of the complexity. 

Open to returning an error prior to the call if Credentials are not set

### CODE REVIEW

File: resources/spaces.go [hunk 1]
```diff
@@ -149,14 +148,19 @@ func Spaces() *schema.Table {
...
 	svc := meta.(*client.Client)
+	log := meta.Logger()
+
 	buckets, err := svc.S3.ListBuckets(ctx, &s3.ListBucketsInput{}, func(options *s3.Options) {
 		options.Region = svc.SpacesRegion
 	})
 	if err != nil {
-		log.Println(err)
+		log.Error("Failed to List Buckets", "Error", err)
+		if !svc.CredentialStatus.Spaces {
```
Comment (roneli) [inline resources/spaces.go:158]: why not check this before doing the call, if its not set the call won't work. Seems you don't check the other API field, so maybe just check if s3 client is nil, imo this should return a diagnostic that you called Spaces API but didn't set client key.

**Review by roneli — CHANGES_REQUESTED** [review]
Agree with the change, I think we can improve a little over the current proposal

File: resources/spaces.go (hunk 1 above)
Comment (bbernays) [reply resources/spaces.go:158]: I put it here incase the bucket was fully public

Comment (yevgenypats) [reply resources/spaces.go:158]: In that case I would do the following:

```
if !svc.CredentialStatus.Spaces {
    log.Warn("Spaces credentials not set. skipping")
    return nil
} else {
   return err
}
```

**Review by yevgenypats — CHANGES_REQUESTED** [review]
looks good one comment

File: client/client.go [hunk 2]
```diff
@@ -21,18 +21,23 @@ const MaxItemsPerPage = 200
...
+	CredentialStatus DoCredentialStruct
+	S3               *s3.Client
 }
 
 func (c *Client) Logger() hclog.Logger {
 	return c.logger
 }
 
+type DoCredentialStruct struct {
+	Api    bool
```
Comment (zagronitay) [inline client/client.go:37]: Not sure if needed
I don't see any usage

Comment (bbernays) [reply client/client.go:37]: I don't think it is, but I had put it in there for completeness.

**Review by zagronitay — APPROVED** [review]
LGTM

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**